### PR TITLE
`migrate-to-v2`: support apps with volumes

### PR DESCRIPTION
**Pull request by alichay**
Use volume forking to migrate apps with volumes.

Creates copies of *every volume in an app*, named "[oldname]_machines", and attaches them to the new machines.
When migrating apps with volumes, there will be a short downtime because nomad has to be scaled down before the volumes can be forked.

For apps without volumes, we maintain zero-downtime migrations because nomad can be scaled down after machines are created.

### CODE REVIEW

**Comment by alichay** [comment]
Blocked on https://github.com/superfly/web/pull/1740

**Review by alichay — COMMENTED** [review]
pointing out some of my questions that I scattered in the codebase:

File: internal/appconfig/setters.go [hunk 1]
```diff
@@ -247,3 +247,15 @@ func (c *Config) SetVolumes(volumes []Volume) {
 		}
 	}
 }
+
+func (c *Config) Volumes() []Volume {
```
Comment (dangra) [inline internal/appconfig/setters.go:251]: totally a nit pick but this is not a setter, we have been placing v1/v2 compatibility getters in config.go mainly.

Comment (alichay) [reply internal/appconfig/setters.go:251]: that's what I get for relying on IDE features and not using my eyeballs. thanks!

**Review by dangra — APPROVED** [review]

**Pull request merged**